commit 350fb97ce64384e59bf9e66a3163f1f0ede65231
Author: Andrew Lewis <nerf@judo.za.org>
Date:   Sat Nov 18 00:07:54 2023 +0200

    [Test] Skip miltertest tests if it is not installed

diff --git a/test/functional/cases/131_dkim_signing/006_milter.robot b/test/functional/cases/131_dkim_signing/006_milter.robot
index cfa1bafce..b51fae795 100644
--- a/test/functional/cases/131_dkim_signing/006_milter.robot
+++ b/test/functional/cases/131_dkim_signing/006_milter.robot
@@ -1,6 +1,7 @@
 *** Settings ***
 Suite Setup     Rspamd Setup
 Suite Teardown  Rspamd Teardown
+Test Tags       miltertest
 Library         Process
 Library         ${RSPAMD_TESTDIR}/lib/rspamd.py
 Resource        ${RSPAMD_TESTDIR}/lib/rspamd.robot
@@ -21,6 +22,7 @@ MULTIPLE SIGNATURES
 *** Keywords ***
 Milter Test
   [Arguments]  ${mtlua}
+  Skip If  not ${HAVE_MILTERTEST}
   ${result} =  Run Process  miltertest  -Dport\=${RSPAMD_PORT_PROXY}  -Dhost\=${RSPAMD_LOCAL_ADDR}  -s  ${RSPAMD_TESTDIR}/lua/miltertest/${mtlua}
   ...  cwd=${RSPAMD_TESTDIR}/lua/miltertest
   Should Match Regexp  ${result.stderr}  ^$
diff --git a/test/functional/cases/180_milter.robot b/test/functional/cases/180_milter.robot
index be9a06466..f7c1385e6 100644
--- a/test/functional/cases/180_milter.robot
+++ b/test/functional/cases/180_milter.robot
@@ -1,6 +1,7 @@
 *** Settings ***
 Suite Setup     Rspamd Setup
 Suite Teardown  Rspamd Teardown
+Test Tags       miltertest
 Library         Process
 Library         ${RSPAMD_TESTDIR}/lib/rspamd.py
 Resource        ${RSPAMD_TESTDIR}/lib/rspamd.robot
@@ -30,6 +31,7 @@ COMBINED TEST
 *** Keywords ***
 Milter Test
   [Arguments]  ${mtlua}
+  Skip If  not ${HAVE_MILTERTEST}
   ${result} =  Run Process  miltertest  -Dport\=${RSPAMD_PORT_PROXY}  -Dhost\=${RSPAMD_LOCAL_ADDR}  -s  ${RSPAMD_TESTDIR}/lua/miltertest/${mtlua}
   ...  cwd=${RSPAMD_TESTDIR}/lua/miltertest
   Should Match Regexp  ${result.stderr}  ^$
diff --git a/test/functional/lib/vars.py b/test/functional/lib/vars.py
index 898d21286..0a12a8191 100644
--- a/test/functional/lib/vars.py
+++ b/test/functional/lib/vars.py
@@ -1,6 +1,8 @@
+import shutil
 import socket
 
 CONTROLLER_ERRORS = True
+HAVE_MILTERTEST = shutil.which('miltertest') and True or False
 RSPAMD_EXTERNAL_RELAY_ENABLED = False
 RSPAMD_KEY_PVT1 = 'ekd3x36tfa5gd76t6pa8hqif3ott7n1siuux68exbkk7ukscte9y'
 RSPAMD_KEY_PUB1 = 'm8kneubpcjsb8sbsoj7jy7azj9fdd3xmj63txni86a8ye9ncomny'
