commit 8d30056088484ad7b00ff0664b8814516729dcc3
Author: Andrew Lewis <nerf@judo.za.org>
Date:   Sat Nov 18 02:08:46 2023 +0200

    [Test] Fix dummy http server on python 3.6

diff --git a/test/functional/util/dummy_http.py b/test/functional/util/dummy_http.py
index 832dbdc79..c1abf7eb8 100755
--- a/test/functional/util/dummy_http.py
+++ b/test/functional/util/dummy_http.py
@@ -141,4 +141,5 @@ async def main():
     await asyncio.Event().wait()
 
 if __name__ == "__main__":
-    asyncio.run(main())
+    loop = asyncio.get_event_loop()
+    loop.run_until_complete(main())
